osm2gtfs/creators/ci_abidjan/trips_creator_ci_abidjan.py

```diff
@@ -0,0 +1,209 @@
+        }
+
+        feed.SetDefaultServicePeriod(default_service_period)
+        # This sets the default agency to the agency in config.json
+        # Not ever used, but needed to avoid an error in feed.AddRoute() below
+        feed.GetDefaultAgency()
+
+        for route_ref, line in sorted(lines.iteritems()):
+            if not isinstance(line, Line):
+                continue
+            # print("Generating schedule for line: " + route_ref)
+
+            if 'operator' in line.tags and line.tags['operator']:
+                agency_id = line.tags['operator']
+                if agency_id == self.config['agency']['agency_name']:
+                    agency_id = self.config['agency']['agency_id']
+            else:
+                agency_id = 'Unknown Agency'
+
+            try:
+                agency = feed.GetAgency(agency_id)
+            except KeyError:
+                agency = self._init_agency(feed, agency_id)
+
+            line_gtfs = feed.AddRoute(
+                short_name=str(line.route_id),
+                long_name=line.name,
+                # we change the route_long_name with the 'from' and 'to' tags
+                # of the route as the route_master name tag contains
+                # the line code (route_short_name)
+                route_type=line.route_type,
+                route_id=line.osm_id)
+            line_gtfs.agency_id = agency.agency_id
+            line_gtfs.route_desc = ""
+            line_gtfs.route_color = "1779c2"
+            line_gtfs.route_text_color = "ffffff"
+
+            route_index = 0
+            itineraries = line.get_itineraries()
+            transport_hours = transporthours.main.Main()
+
+            line_hours_list = transport_hours.tagsToGtfs(line.tags)
+            line_hours_dict = self._process_transport_hours(feed, line_hours_list)
+
+            for a_route in itineraries:
+                itinerary_hours_list = transport_hours.tagsToGtfs(a_route.tags)
+
+                if itinerary_hours_list:
+                    itinerary_hours_dict = self._process_transport_hours(feed, itinerary_hours_list)
+                elif line_hours_dict:
+                    itinerary_hours_dict = line_hours_dict
+                else:
+                    itinerary_hours_dict = default_hours_dict
+
+                for service_id,itinerary_hours in itinerary_hours_dict.items():
+                    service_period = feed.GetServicePeriod(service_id)
+                    trip_gtfs = line_gtfs.AddTrip(feed, service_period=service_period)
+                    trip_gtfs.shape_id = self._add_shape_to_feed(feed, a_route.osm_id, a_route)
+                    trip_gtfs.direction_id = route_index % 2
+                    route_index += 1
+
+                    if a_route.fr and a_route.to:
+                        trip_gtfs.trip_headsign = a_route.to
+                        line_gtfs.route_long_name = a_route.fr + " ↔ ".decode('utf-8') +  a_route.to
```

Comment:
As we don't always (will) have a route_short_name, it is not always needed to change the route_long_name. We should now check if route_short_name is in route_long_name to define if we keep it that way or if we want to make a new one with from and to OSM tags.

Reply:
Je ne comprends pas tout à fait ce que tu veux. Peux-tu clarifier ? E


n PJ les routes avec route_long_name pris directement des données OSM, et puis une autre version où les route_long_name sont renommés sur la base de from et to. D'ailleurs, il y a erreur dans les donnes OSM. Le nom des lignes 10179435 et 10184139 est le même dans les données OSM : "gbaka : Adjamé Liberté ↔ Abobo Gare Mairie". Mais selon les champs to/from des itinéraires de 10179435, sont nom devrait être "gbaka : Adjamé Marché Gouro ↔ Abobo Gare Mairie".

Reply:
La raison pour laquelle on utilise `from` ↔ `to` au lieu de `name` pour `route_long_name` est parce que le `name` comprend en général le numéro de la ligne, qu'on met aussi dans `route_short_name`. Hors, GTFS Best Practices recommande de ne pas répéter le `route_short_name` dans le `route_long_name` : https://gtfs.org/best-practices/#routestxt

Mais, pour les lignes qui n'ont pas de numéro (`ref`), autant garder le `name` complet, qui est plus précis, puisqu'il détaille le type de véhicule informel qui réalise la course, une chose utile au voyageur mais que le GTFS ne permet pas encore de renseigner.